catchup/service.go

```diff
@@ -165,7 +177,7 @@ func (s *Service) innerFetch(r basics.Round, peer network.Peer) (blk *bookkeepin
 	go func() {
 		select {
 		case <-stopWaitingForLedgerRound:
-		case <-s.ledger.Wait(r):
+		case <-ledgerWaitCh:
```

Comment:
Ideally, don't we want errLedgerAlreadyHasBlock returned when ledgerWaitCh fires?

Reply:
Sounds like a good idea. I'll make it so.